include/counter.h

```diff
@@ -100,26 +250,86 @@ static inline u32_t _impl_counter_read(struct device *dev)
 
 /**
  * @brief Set an alarm.
- * @param dev Pointer to the device structure for the driver instance.
- * @param callback Pointer to the callback function. If this is NULL,
- *                 this function unsets the alarm.
- * @param count Number of counter ticks. This is relative value, meaning
- *              an alarm will be triggered count ticks in the future.
- * @param user_data Pointer to user data.
+ *
+ * In case of absolute request, maximal value that can be set is equal to
+ * wrap value set by counter_set_wrap call or default, maximal one. in case of
+ * relative request, Maximal value can be retrieved using
+ * counter_get_max_relative_alarm.
+ *
+ * @param dev		Pointer to the device structure for the driver instance.
+ * @param alarm_cfg	Alarm configuration.
  *
  * @retval 0 If successful.
- * @retval -ENOTSUP if the counter was not started yet.
- * @retval -ENODEV if the device doesn't support interrupt (e.g. free
- *                        running counters).
- * @retval Negative errno code if failure.
+ * @retval -ENOTSUP if request is not supported (device does not support
+ *		    interrupts or requested channel) or the counter was not
+ *		    started yet.
```

Comment:
I was wondering if this was really necessary to have the counter running before setting an alarm. Maybe it should be permitted to first configure the counter completely, i.e. set a wrap value for it and some alarms, and then start it?

Reply:
yes, i agree with you. I will update this comment.